src/github/mod.rs

```diff
@@ -195,6 +202,55 @@ impl SyncGitHub {
         Ok(diffs)
     }
 
+    // collect all org and respective teams members in a HashMap
+    fn map_teams_to_org(&self) -> anyhow::Result<HashMap<String, HashSet<u64>>> {
+        let mut org_team_members: HashMap<String, HashSet<u64>> = HashMap::new();
+
+        for team in &self.teams {
+            let mut team_org;
+
+            if let Some(gh) = &team.github {
+                for toml_gh_team in &gh.teams {
+                    team_org = toml_gh_team.org.clone();
+                    let toml_team_mems_gh_id: HashSet<u64> =
+                        toml_gh_team.members.iter().copied().collect();
+
+                    org_team_members
+                        .entry(team_org)
+                        .or_default()
+                        .extend(toml_team_mems_gh_id);
+                }
+            }
+        }
+        Ok(org_team_members)
+    }
+
+    // create diff against github org members against toml team members
+    fn diff_teams_gh_org(
+        &self,
+        org_team_members: HashMap<String, HashSet<u64>>,
+    ) -> anyhow::Result<OrgMembershipDiff> {
+        let mut org_with_members_to_be_removed: HashMap<String, HashSet<String>> = HashMap::new();
+
+        for (gh_org, toml_members_across_teams) in org_team_members.into_iter() {
+            let gh_org_members = self.org_members.get(&gh_org).unwrap();
+
+            let members_to_be_removed = (&toml_members_across_teams - gh_org_members)
```

Comment:
It's the other way around, we want to remove users (from GitHub) that are currently on GitHub, but are not in the TOML teams.
```suggestion
            let members_to_be_removed = (gh_org_members - &toml_members_across_teams)
```